### PR TITLE
Apply fading effect on DetailPage for MasterDetailPage on iOS

### PR DESCRIPTION
**Pull request by akamud**
### Description of Change ###

Something that bugs me is the fact that iOS's MasterDetailPage doesn't fade the Detail page when opening the Menu. This leads to a bad UX, since both the menu and the detail are shown at the same z-axis level, with no "background feel" applied to the Detail page. 
Android's implementation doesn't suffer from this, **neither does iOS Shell implementation**.

Once I noticed Shell "fixed" this behavior, I wanted to do the same for the MasterDetailPage on Xamarin.Forms for iOS.

So this PR pretty much uses the same logic used in iOS Shell to fade the Detail page when MasterDetail menu is opened.

### Platforms Affected ### 

- iOS

### Behavioral/Visual Changes ###

The visual rendering for MasterDetailPage would change for every Xamarin.Forms application.

I have 2 specific points that I would like to get a review:
* Since this would be a visual breaking change, maybe we could hide this behind a PlatformSpecific?
* If no background color is provided for the detail page, a white background is used, this gives the impression that the detail page is clearing instead of going to the background. Because of this I am forcing a "black fade", this could cause problems if someone is changing the background color of the Detail page for some reason. Should we suggest on documentation that people can put a black background color if they want this behavior? Maybe we can expose a way of customizing this specific color with another property?

#### Before
before

#### After
after

### Testing Procedure ###
Open any MasterDetailPage (like the ControlGallery) and play with the Menu.

Not sure how I could write an automated test for this.

### PR Checklist ###

- [ ] Has automated tests 
- [ ] Rebased on top of the target branch at time of PR
- [ ] Changes adhere to coding standard

### CODE REVIEW

**Comment by VladislavAntonyuk** [comment]
I think it should be controlled by flag IsFadingEffectEnabled or we should have a possibility to set Opacity

**Comment by akamud** [comment]
Totally agree that this needs a PlatformSpecific because of the breaking change. I had an implementation using an effect and one using a custom renderer, but I couldn't attach myself in the middle of animations to provide a fluid experience every time, I was stuck with some visual glitches because of that, that's why I decided to send the PR.

I thought Xamarin's team had to define the name of the PlatformSpecific because of the impact of this. Can I just push changes using a name I pick?

**Comment by rmarinho** [comment]
@akamud did you got some time to get this as a PS ?

**Comment by akamud** [comment]
Sorry for taking so long. I'll work on this over the weekend.

**Comment by rmarinho** [comment]
ping @akamud :)

**Comment by akamud** [comment]
I apologize for the delay, I finally had the time to make the changes proposed. Pushed new changes using a Platform specific and added a sample in the MasterDetailPageiOS from the gallery.

File: Xamarin.Forms.Platform.iOS/Renderers/PhoneMasterDetailRenderer.cs [hunk 1]
```diff
@@ -265,12 +274,16 @@ void LayoutChildren(bool animated)
 				UIView.BeginAnimations("Flyout");
 				var view = _detailController.View;
 				view.Frame = target;
+				detailView.Layer.Opacity = (float)opacity;
```
[the remark this answers is not in the document]
Comment (akamud) [reply Xamarin.Forms.Platform.iOS/Renderers/PhoneMasterDetailRenderer.cs]: This needs to be inside the if statement so it animates the fade. If I set it before the if statement you will see this glitch:

glitch

**Comment by akamud** [comment]
Is there anything else I must do on this?

**Comment by rmarinho** [comment]
@akamud needs rebase, it's failing build on OSX

**Comment by akamud** [comment]
Done.

**Comment by akamud** [comment]
I see the Windows step is failing. Is there something I should do?

**Comment by rmarinho** [comment]
@akamud do you mind rebase again, i think it's a issue we found that is nos fixed since i merged the branchs up this morning

**Comment by akamud** [comment]
Is this going to be added to the documentation?

**Pull request merged**